private/bufpkg/bufplugin/bufplugin.go

```diff
@@ -241,59 +255,73 @@ func ParseTemplateVersionConfig(config string) (*TemplateVersionConfig, error) {
 	return templateVersionConfig, nil
 }
 
-// PluginResult describes the result of executing a plugin.
-type PluginResult struct {
-	Plugin   PluginVersion
-	Response *pluginpb.CodeGeneratorResponse
+// File represents a generated file
+type File struct {
+	Name    string
+	Content []byte
+}
+
+// MergedPluginResult holds the files for a plugin result.
+type MergedPluginResult struct {
+	Files []*File
 }
 
 // MergeInsertionPoints traverses the plugin results and merges any insertion points present in any responses.
-// The order of inserts depends on the order of the input plugins.
-func MergeInsertionPoints(ctx context.Context, pluginResults []PluginResult) (map[string][]byte, error) {
-	files := make(map[string][]byte)
-	for _, pluginResult := range pluginResults {
-		if pluginResult.Response.Error != nil {
-			return nil, fmt.Errorf("unexpected response error: %s", *pluginResult.Response.Error)
+// The order of inserts depends on the order of the input plugins. The returned results are in the same order
+// as the input plugin results.
+func MergeInsertionPoints(responses []*pluginpb.CodeGeneratorResponse) ([]MergedPluginResult, error) {
```

Comment:
For consistency with pointers elsewhere, this could be a `[]*MergedPluginResult`.

Reply:
Good shout